src/p5.dance.js

```diff
@@ -455,6 +455,24 @@ module.exports = class DanceParty {
 
   // Dance Moves
 
+  alternateMoves(group, n, move1, move2) {
+    group = this.getGroupByName_(group);
+    let currentMeasure = Math.floor(this.getCurrentMeasure());
+    if (currentMeasure === 0) {
+      currentMeasure = 1;
+    }
+    group.forEach(sprite => {
+      this.changeMoveLR(sprite, move1, -1);
+      sprite.alternatingMoveInfo = {
+        start: currentMeasure,
+        cadence: n,
+        move1: move1,
+        move2: move2,
+        current: 1
```

Comment:
Is this necessary or could this be calculated from the start, cadence, and current measure?

Reply:
I don't think so (but I could be wrong so please double-check my logic here!)
You can calculate from start, cadence, and current what the dance _should_ be. But this is keeping track of what the current dance _is_. This is so that we can call `this.changeMoveLR` only when the dance needs to change (ie. when what the dance should be and what the dance is are not the same).
This check is happening inside the draw loop (roughly 50x per measure), so we don't want to call `this.changeMoveLR` every time, only when the move needs to change